app/models/user.rb

```diff
@@ -252,12 +253,43 @@ def last_activities(last_activity_id = nil, per_page = 10)
       .uniq
   end
 
+  def self.find_by_valid_wopi_token(token)
+    Rails.logger.warn "WOPI: searching by token #{token}"
+    User
+    .joins("LEFT OUTER JOIN tokens ON user_id = users.id")
```

Comment:
`.joins` method makes an inner join, so yeah - if outer join is not needed, use `.joins`, otherwise leave it as it is.